CharlieBackend.Business/Services/LessonService.cs

```diff
@@ -121,6 +122,24 @@ public async Task<Result<IList<LessonDto>>> GetAllLessonsForMentor(long mentorId
             return Result<IList<LessonDto>>.GetSuccess(_mapper.Map<IList<LessonDto>>(lessons));
         }
 
+        public async Task<Result<IList<LessonDto>>> GetLessonsForMentorAsync(FilterLessonsRequestDto filterModel, ClaimsPrincipal userContext)
```

Comment:
Also you don't need to use Result for this service. It would be enough to return IList<LessonDto>